test/helpers/lsp.cc

```diff
@@ -208,7 +208,7 @@ void checkServerCapabilities(const ServerCapabilities &capabilities) {
     // We don't support all possible features. Make sure we don't make any false claims.
     CHECK(capabilities.definitionProvider.value_or(false));
     CHECK(capabilities.typeDefinitionProvider.value_or(false));
-    CHECK_FALSE(capabilities.implementationProvider.has_value());
+    CHECK(capabilities.implementationProvider.has_value());
```

Comment:
We should be asserting that this is `true` (we enable all experimental features in tests):

```suggestion
    CHECK(capabilities.implementationProvider.value_or(false));
```